Model/Api/UniversalWebhook.php

```diff
@@ -0,0 +1,176 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2017-2020 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Model\Api;
+
+use Bolt\Boltpay\Api\UniversalWebhookInterface;
+use Bolt\Boltpay\Api\OrderManagementInterface;
+use Bolt\Boltpay\Api\Data\UniversalWebhookResultInterface as Result;
+use Bolt\Boltpay\Exception\BoltException;
+use Bolt\Boltpay\Helper\Bugsnag;
+use Bolt\Boltpay\Helper\Log as LogHelper;
+use Bolt\Boltpay\Model\ErrorResponse as BoltErrorResponse;
+use Magento\Framework\Webapi\Rest\Response;
+
+class UniversalWebhook implements UniversalWebhookInterface
+{
+
+    /**
+     * @var OrderManagementInterface
+     */
+    private $orderManagement;
+
+    /**
+     * @var Result
+     */
+    private $result;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+
+    /**
+     * @var LogHelper
+     */
+    private $logHelper;
+
+    /**
+     * @var BoltErrorResponse
+     */
+    protected $errorResponse;
+
+    /**
+     * @var Response
+     */
+    private $response;
+
+    public function __construct(
+        OrderManagementInterface $orderManagement,
+        Result $result,
+        Bugsnag $bugsnag,
+        LogHelper $logHelper,
+        BoltErrorResponse $errorResponse,
+        Response $response
+    )
+    {
+        $this->orderManagement = $orderManagement;
+        $this->result = $result;
+        $this->bugsnag = $bugsnag;
+        $this->logHelper = $logHelper;
+        $this->errorResponse = $errorResponse;
+        $this->response = $response;
+    }
+
+    /**
+     * @api
+     * 
+     * @param string $type
+     * @param string $object
+     * @param mixed $data
+     * 
+     * @return boolean
+     */
+    public function execute(
+        $type = null,
+        $object = null,
+        $data = null
+    )
+    {
+        try {
+            // This just reuses the existing manage function by pulling the necessary data from the request.
+            $this->orderManagement->manage(
```

Comment:
The purpose of the orderManagement class is to handle the specific API endpoint.
If we need the same code for handling two different API endpoints I'd rather prefer to move the code to the third class and call it from two API classes then call one API from another.

Reply:
While I don't disagree with your point here, this is already what's been done in https://github.com/BoltApp/bolt-magento2/pull/1085 now: calling existing handler methods from the universal handler. Do you think this is something that could be scoped as a refactor after getting the MVP version of the universal webhook and api out? Or should this block merging altogether and we set up tickets to refactor the universal API stuff as well?

Reply:
I am ok to merge this, please add task to the backlog for refactoring